src/Lecture2.hs

```diff
@@ -74,7 +81,15 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+
+removeAt :: Eq a => Int -> [a]-> (Maybe a, [a])
+removeAt idx xs 
+  | idx < 0           = (Nothing, xs)
+  | drop idx xs == [] = (Nothing, xs)  
```

Comment:
I think this check is redundant. You can pattern match on `back` after calling `splitAt` instead to see if you need to return `Nothing` or `Just smth`. If you remove this equality check, you'll improve performance because you now avoid traversing the list prefix twice. And you also can remove the `Eq` constraint from the type signature which becomes redundant after this refactoring 🙂